devdoc/sm_requirements.md

```diff
@@ -0,0 +1,228 @@
+# State manager requirements
```

Comment:
I have waited a long time to see this done, so I are very excited. My starting this effort is from 1y ago!